applications/callflow/src/module/cf_voicemail.erl

```diff
@@ -539,27 +544,33 @@ record_voicemail(AttachmentName, #mailbox{max_message_length=MaxMessageLength
         {'ok', Msg} ->
             Length = kz_json:get_integer_value(<<"Length">>, Msg, 0),
             IsCallUp = kz_json:get_value(<<"Hangup-Cause">>, Msg) =:= 'undefined',
-            case IsCallUp
-                andalso review_recording(AttachmentName, 'true', Box, Call)
-            of
-                'false' ->
-                    new_message(AttachmentName, Length, Box, Call);
-                {'ok', 'record'} ->
-                    record_voicemail(tmp_file(Ext), Box, Call);
-                {'ok', _Selection} ->
-                    cf_util:start_task(fun new_message/4, [AttachmentName, Length, Box], Call),
-                    _ = kapps_call_command:prompt(<<"vm-saved">>, Call),
-                    _ = kapps_call_command:prompt(<<"vm-thank_you">>, Call),
-                    'ok';
-                {'branch', Flow} ->
-                    _ = new_message(AttachmentName, Length, Box, Call),
-                    _ = kapps_call_command:prompt(<<"vm-saved">>, Call),
-                    {'branch', Flow}
-            end;
+            Digit = kz_json:get_value(<<"Terminator">>, Msg),
+            maybe_review_recording(AttachmentName, Box, Call, Digit, Length, IsCallUp);
         {'error', _R} ->
             lager:info("error while attempting to record a new message: ~p", [_R])
     end.
 
+-spec maybe_review_recording(ne_binary(), mailbox(), kapps_call:call(), ne_binary() | 'undefined', integer(), boolean()) -> 'ok'.
+maybe_review_recording(AttachmentName, #mailbox{}=Box, Call, _Digit, Length, _IsCallUp='false') ->
+    new_message(AttachmentName, Length, Box, Call);
+maybe_review_recording(_AttachmentName, #mailbox{keys=#keys{continue=Digit}}, _Call, Digit, _Length, _IsCallUp) ->
+    lager:info("caller chose to continue to the next element in the callflow"),
+    'ok';
```

Comment:
Why not saving the already recorded voicemail and then continue?
Also no need for 'ok'.